### PR TITLE
Post feedback update lesson 4

### PR DESCRIPTION
**Pull request by ClementineCttn**
Following our collective feedback on the 3rd GDCU workshop, I have amended lesson 4 mostly to make the leaflet challenge easier, to emphasize replicability and save map as images as the end of the main exercise.

### CODE REVIEW

File: episodes/19-basic-gis-with-r-sf.Rmd [hunk 1]
```diff
@@ -69,7 +71,9 @@ buildings$start_date <- as.numeric(buildings$start_date)
 old_buildings <- buildings %>%
   filter(start_date <= old)
 
- ggplot(data = old_buildings) + geom_sf(colour="red")
+ ggplot(data = old_buildings) + 
+   geom_sf(colour="red") +
+     coord_sf(datum = st_crs(28992))
```
Comment (cforgaci) [inline episodes/19-basic-gis-with-r-sf.Rmd:76]:
```suggestion
   coord_sf(datum = st_crs(28992))
```

File: episodes/18-import-and-visualise-osm-data.Rmd [hunk 2]
```diff
@@ -187,22 +195,58 @@ buildings$build_date <- if_else(start_date < 1900, 1900, start_date)
  ggplot(data = buildings) +
    geom_sf(aes(fill = build_date, colour=build_date))  +
    scale_fill_viridis_c(option = "viridis")+
-   scale_colour_viridis_c(option = "viridis")
+   scale_colour_viridis_c(option = "viridis") +
+     coord_sf(datum = st_crs(28992))
```
Comment (cforgaci) [inline episodes/18-import-and-visualise-osm-data.Rmd:199]:
```suggestion
   coord_sf(datum = st_crs(28992))
```

File: episodes/18-import-and-visualise-osm-data.Rmd [hunk 3]
```diff
@@ -187,22 +195,58 @@ buildings$build_date <- if_else(start_date < 1900, 1900, start_date)
...
+  st_transform(.,crs=28992)
+start_date <- as.numeric(buildings$start_date)
+buildings$build_date <- if_else(start_date < year, year, start_date)
+ ggplot(data = buildings) +
+   geom_sf(aes(fill = build_date, colour=build_date))  +
+   scale_fill_viridis_c(option = "viridis")+
+   scale_colour_viridis_c(option = "viridis") +
+   ggtitle(paste0("Old buildings in ",cityname)) +
+   coord_sf(datum = st_crs(28992))
+}
```
Comment (cforgaci) [inline episodes/18-import-and-visualise-osm-data.Rmd:228]:
```suggestion
extract_buildings <- function(cityname, year=1900){
  nominatim_polygon <- geo_lite_sf(address = cityname, points_only = FALSE)
  bb <- st_bbox(nominatim_polygon)
  
  x <- opq(bbox = bb) %>%
     add_osm_feature(key = 'building') %>%
     osmdata_sf()
     
  buildings <- x$osm_polygons %>%
    st_transform(.,crs=28992)
    
  start_date <- as.numeric(buildings$start_date)
  
  buildings$build_date <- if_else(start_date < year, year, start_date)
   ggplot(data = buildings) +
     geom_sf(aes(fill = build_date, colour=build_date))  +
     scale_fill_viridis_c(option = "viridis")+
     scale_colour_viridis_c(option = "viridis") +
     ggtitle(paste0("Old buildings in ",cityname)) +
     coord_sf(datum = st_crs(28992))
}
```

File: episodes/18-import-and-visualise-osm-data.Rmd [hunk 4]
```diff
@@ -187,22 +195,58 @@ buildings$build_date <- if_else(start_date < 1900, 1900, start_date)
...
 
 ## Challenge: import an interactive basemap layer under the buildings with `Leaflet` (20min)
 
-- Check out the [leaflet package documentation](https://rstudio.github.io/leaflet/)
-- Plot a basemap in Leaflet and try different tiles in the [basemap documentation](https://rstudio.github.io/leaflet/basemaps.html)
-- Transform the buildings into WGS84 projection and add them to the basemap layer with the `addPolygons()` function.
-- Have the `fillColor` of these polygons represent the `build_date` variable. See the [choropleth documentation](https://rstudio.github.io/leaflet/choropleths.html) for use of colors. Tip: use the examples given in the documentation and replace the variable names where needed.
+Leaflet is a ["open-source JavaScript library for mobile-friendly interactive maps"](https://leafletjs.com/). Within R, the `leaflet` package allows you to build such interactive maps. As with `ggplot2`, you build a map with a collection of layers. In this case, you will have the leaflet basemap, some tiles, and shapes on top (such as markers, polygons, etc.). 
+
+- Check out the [leaflet package documentation](https://rstudio.github.io/leaflet/) and GDCU cheatsheet.
```
Comment (cforgaci) [inline episodes/18-import-and-visualise-osm-data.Rmd:244]: We need to add a link to the cheatsheet here.

File: episodes/18-import-and-visualise-osm-data.Rmd [hunk 5]
```diff
@@ -187,22 +195,58 @@ buildings$build_date <- if_else(start_date < 1900, 1900, start_date)
...
+- Plot a basemap using `leaflet`
+- Add a layer of tiles, for instance provider tiles [basemap documentation](https://rstudio.github.io/leaflet/basemaps.html)
+- Transform the buildings into WGS84 projection
+- Add the building layer to your leaflet map using the `addPolygons()` function.
+- Use the `fillColor` of these polygons represent the `build_date` variable. See the [choropleth documentation](https://rstudio.github.io/leaflet/choropleths.html) DCU cheatsheet for how to use of colors in polygons.
```
Comment (cforgaci) [inline episodes/18-import-and-visualise-osm-data.Rmd:249]: You mean `GDCU`, right? Link to be added here too.

**Review by cforgaci — COMMENTED** [review]
Looks good, @ClementineCttn! I left a couple of tiny suggestions. If you process them, we can merge the PR.

File: episodes/18-import-and-visualise-osm-data.Rmd [hunk 6]
```diff
@@ -187,22 +195,58 @@ buildings$build_date <- if_else(start_date < 1900, 1900, start_date)
...
+extract_buildings("Brielle, NL")
+
+#test on Naarden
+extract_buildings("Naarden, NL")
+
+```
 
 ::::::::::::::::::::::::::::::::::::: challenge 
 
 ## Challenge: import an interactive basemap layer under the buildings with `Leaflet` (20min)
```
Comment (cforgaci) [inline episodes/18-import-and-visualise-osm-data.Rmd:240]:
```suggestion
## Challenge: import an interactive basemap layer under the buildings with Leaflet (20min)
```

**Comment by cforgaci** [comment]
@ClementineCttn when you finalise your lesson can you please do the checks from #7 as well?

**Comment by cforgaci** [comment]
@ClementineCttn, please consider #31 in your updates as well.

**Comment by ClementineCttn** [comment]
All of Claudiu's comment have been resolved here and in the new small PR, including the removal of nominatimlite package, naming code chunks, etc.

**Pull request merged**